Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/app/app.component.ts

```diff
@@ -13,20 +15,30 @@ export class AppComponent {
   weather = 50;
   speed = 0;
   battery = 0;
+  range = 0;
+  speedInput = new FormControl(null, [Validators.required, Validators.min(0), Validators.max(90)]);
+  batteryInput = new FormControl(null, [Validators.required, Validators.min(0), Validators.max(100)]);
+  outputMessage = ""
+  buttonPress = false;
 
   onMouseWeather(event: MouseEvent) {
     this.weather = parseInt((event.target as HTMLInputElement).value);
+    this.buttonPress = false;
   }
 
   onKeySpeed(event: KeyboardEvent) {
     this.speed = parseInt((event.target as HTMLInputElement).value);
+    this.buttonPress = false;
   }
 
   onKeyBattery(event: KeyboardEvent) {
     this.battery = parseInt((event.target as HTMLInputElement).value);
+    this.buttonPress = false;
   }
 
   calculateRange() {
-
-  }
+    this.buttonPress = true;
+    this.range = -(this.speed * this.speed * this.battery / 2500) + (4 * this.battery) + this.weather;
+    this.outputMessage = "The predicted range of the Elysia is " + this.range + "km.";
 }
```

Comment:
you should add an indent here. (right now its the same indentation as the line before)